icu4c/source/i18n/calendar.cpp

```diff
@@ -1532,13 +1532,13 @@ void Calendar::computeFields(UErrorCode &ec)
     U_ASSERT(fFields[UCAL_DST_OFFSET] <= getMaximum(UCAL_DST_OFFSET));
 }
 
-uint8_t Calendar::julianDayToDayOfWeek(double julian)
+uint8_t Calendar::julianDayToDayOfWeek(int32_t julian)
 {
     // If julian is negative, then julian%7 will be negative, so we adjust
     // accordingly.  We add 1 because Julian day 0 is Monday.
-    int8_t dayOfWeek = (int8_t) uprv_fmod(julian + 1, 7);
+    int8_t dayOfWeek = (int8_t) ((julian + 1) % 7);
 
-    uint8_t result = (uint8_t)(dayOfWeek + ((dayOfWeek < 0) ? (7+UCAL_SUNDAY ) : UCAL_SUNDAY));
+    uint8_t result = (uint8_t)(dayOfWeek + (7*(dayOfWeek < 0)+UCAL_SUNDAY));
```

Comment:
I've reverted this change with the other bool changes. I'm actually less sure about this one than the other optimizations since here we still do a comparison and a multiplication. This simplifies the calendar changes to removing `fmod` (which had the biggest performance impact) and the unnecessary casts.

Reply:
That sounds good.